code/modules/mob/living/carbon/slime/slime.dm

```diff
@@ -894,6 +894,87 @@ mob/living/carbon/slime/var/temperature_resistance = T0C+75
 	icon_state = "bottle14"
 	w_class = W_CLASS_TINY
 
+/obj/item/weapon/slimechange
+	name = "slime dye"
+	desc = "A bottle of dye that can be used on a slime to change it's color."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle14"
+	w_class = W_CLASS_TINY
+
+/obj/item/weapon/slimechange/attack(var/mob/living/M, mob/user as mob)
+	if(istype(M, /mob/living/carbon/slime))
+		var/mob/living/carbon/slime/OS = M
+		if(OS.stat)
+			to_chat(user, "<span class='warning'>You're unable to use the [src] on the dead slime!</span>")
+			return..()
+		var/new_color = input(user, "Choose a new color for the dye:", "Color Select") as anything in list(
+			//"adamantine",
+			//"black",
+			"blue",
+			//"bluespace",
+			//"cerulean",
+			"darkpurple",
+			"darkblue",
+			"darkpurple",
+			"gold",
+			"grey",
+			//"lightpink",
+			"metal",
+			//"oil",
+			"orange",
+			"pink",
+			"purple",
+			//"pyrite",
+			"red",
+			//"sepia",
+			"silver",
+			"yellow")
+		if(new_color && !OS.stat)
```

Comment:
All sorts of things could have happened between this and the line above, and you're checking for none of them. For instance: The two mobs might not be adjacent anymore, the user might not exist anymore, be dead or incapacitated, the target might not exist anymore, the slime dye itself could have been used up, and so on.

Reply:
If you could please post your endless list of hypotheticals that would be great. I will add checks for the stuff you mentioned but I don't know every single possible state that they could be in. 
As for the adjacency check one I don't think that one should be required since you're dealing with simple mobs that can move around a whole lot. None of the other slime potions look for it by the end.

Reply:
You could fix the ones I pointed out for a start.
The lack of the adjacency check IS an issue, it makes no sense that I am able to telepathically force a slime to change a slime because at some point in the past I smacked it with the dye.